### PR TITLE
Add a contributiong guide

### PR DESCRIPTION
**Pull request by Ketouem**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Contributing to Nyuki
```
Comment (lerignoux) [inline CONTRIBUTING.md]: should we mention this file in the README ?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+* [Code of conduct](#code-of-conduct)
+* [Issues and bug reports](#issues-and-bug-reports)
+* [New feature proposals](#new-feature-proposals)
+* [Patches submission](#patches-submission)
+
+## Code of conduct
+
+(Adapated from [TODO Group's Open Source of Conduct](http://todogroup.org/opencodeofconduct/))
+
+This code of conduct outlines our expectations for participants within the Surycat community. We are committed to providing a welcoming and inspiring community for all and expect our code of conduct to be honored. Anyone who violates this code of conduct may be banned from the community.
```
Comment (lerignoux) [inline CONTRIBUTING.md]: to provide ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+* **Be considerate**: Your work will be used by other people, and you in turn will depend on the work of others. Any decision you take will affect users and colleagues, and you should take those consequences into account when making decisions.
+* **Try to understand why we disagree**: Disagreements, both social and technical, happen all the time. It is important that we resolve disagreements and differing views constructively. Remember that we’re different.
+* **Be respectful**: Not all of us will agree all the time, but disagreement is no excuse for poor behavior and poor manners.
+* **Be careful in the words that we choose**: we are a community of professionals, and we conduct ourselves professionally. Be kind to others. Do not insult or put down other participants. Harassment and other exclusionary behavior aren’t acceptable.
+
+If you experience or witness unacceptable behavior—or have any other concerns—please report it by contacting us via <abuse@surycat.com>. All reports will be handled with discretion.
```
Comment (lerignoux) [inline CONTRIBUTING.md]: no spaces before the - ?

Comment (lerignoux) [inline CONTRIBUTING.md]: did you checked with @ludo if the address leads somewhere ?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+Nyuki follows the classic GitHub's zen "fork-branch-PR" regarding patch submissions.
+
+1. [Fork](https://help.github.com/articles/fork-a-repo/) Nyuki
+
+    Create a "copy" of this repository under your own username by using the fork feature, this will allow you to experiment your hacks without altering the original project.
+
+2. [Create a branch](https://help.github.com/articles/creating-and-deleting-branches-within-your-repository/)
+
+    When creating a branch, please do include the issue number that your code will address and a sensible name e.g. if an issue has been raised with the index `42` regarding fixing unit tests, the following branch names can be considered as valid:
```
Comment (lerignoux) [inline CONTRIBUTING.md]: github issue number

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+    * johndoe/issue-42/fixing-unit-tests
+    * 42-fixing-tests
+
+
+3. Work on the bug resolution or new feature implementation
+
+    We recommend creating a virtualenv beforehand when hacking on Nyuki. Please refer to the [README file](README.md) for further information regarding installation and set-up of the library.
+
+    To maintain a coherent docing style, any new code must be as [PEP 8](https://www.python.org/dev/peps/pep-0008/) compliant as possible.
```
Comment (lerignoux) [inline CONTRIBUTING.md]: docing ? coding maybe

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+    When creating a branch, please do include the issue number that your code will address and a sensible name e.g. if an issue has been raised with the index `42` regarding fixing unit tests, the following branch names can be considered as valid:
+
+    * johndoe/issue-42/fixing-unit-tests
+    * 42-fixing-tests
+
+
+3. Work on the bug resolution or new feature implementation
+
+    We recommend creating a virtualenv beforehand when hacking on Nyuki. Please refer to the [README file](README.md) for further information regarding installation and set-up of the library.
```
Comment (lerignoux) [inline CONTRIBUTING.md]: there is nothing on that in the README

File: CONTRIBUTING.md (hunk 3 above)
Comment (Ketouem) [inline CONTRIBUTING.md]: @lerignoux actually it's not a - but an [em dash](http://www.thepunctuationguide.com/em-dash.html) and in this case you don't have to put spaces around it.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Ketouem) [inline CONTRIBUTING.md]: :+1:

File: CONTRIBUTING.md (hunk 6 above)
Comment (Ketouem) [inline CONTRIBUTING.md]: Done, there's a quick note at the end of the README.

File: CONTRIBUTING.md (hunk 3 above)
Comment (lerignoux) [inline CONTRIBUTING.md]: ok thanks for the info

File: CONTRIBUTING.md (hunk 6 above)
Comment (lerignoux) [inline CONTRIBUTING.md]: ?? no nothing on a possible virtual env !

**Pull request merged**